### PR TITLE
Release 5.0.0

### PR DESCRIPTION
**Pull request by rlepinski**
Changes:
- Install unity-plugin.aar from a local maven repo instead of bintray. This should make it easier to develop.
- Update iOS to 9.1.0
- Updated Android to 9.2.0 and FCM

FCM Support required processing the google-services.json and generating an xml resource file. Most of the code for GoogleJson.cs it just serializer classes to parse the config.

### CODE REVIEW

File: Assets/UrbanAirship/Editor/GoogleJson.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+			this.config = config;
+		}
+
+		public static GoogleJson FromPath(string path) {
+			string json = System.IO.File.ReadAllText(path);
+			Config config = JsonUtility.FromJson<Config>(json);
+			return new GoogleJson(config);
+		}
+
+		public void WriteXml(string path) {
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/GoogleJson.cs]: This is the conversion code

File: Assets/UrbanAirship/Editor/GoogleJson.cs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+				}
+
+				xmlWriter.WriteEndElement();
+				xmlWriter.WriteEndDocument();
+			}
+		}
+	}
+
+	[System.Serializable]
+	public class Config {
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/GoogleJson.cs]: The rest of the class can be ignored.

File: Assets/UrbanAirship/Editor/UAConfig.cs [hunk 3]
```diff
@@ -34,9 +34,9 @@ public enum LogLevel
 		private static readonly string filePath = "ProjectSettings/UrbanAirship.xml";
 		private static UAConfig cachedInstance;
 
-		static UAConfig ()
+		static UAConfig()
 		{
-		    LoadConfig (). Apply ();
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAConfig.cs]: Fixes a new unity error

File: Assets/UrbanAirship/Editor/UAConfig.cs [hunk 4]
```diff
@@ -203,6 +212,12 @@ public void Migrate()
 				ProductionFCMSenderId = GCMSenderId;
 				GCMSenderId = null;
 			}
+
+			if (Version == null) {
+				GenerateGoogleJsonConfig = true;
+			}
+
+			Version = PluginInfo.Version;
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAConfig.cs]: Need the version to perform some migration

File: Assets/UrbanAirship/Editor/UAUpdater.cs [hunk 5]
```diff
@@ -1,82 +1,76 @@
 /*
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAUpdater.cs]: White space + copyright. Ignore this file

File: Assets/UrbanAirship/Editor/UAUtils.cs [hunk 6]
```diff
@@ -1,17 +1,15 @@
 /*
- Copyright 2017 Urban Airship and Contributors
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAUtils.cs]: White space + copyright. Ignore this file

File: Assets/UrbanAirship/Platforms/Android/UAirshipPluginAndroid.cs [hunk 7]
```diff
@@ -1,168 +1,155 @@
 /*
```
Comment (rlepinski) [inline Assets/UrbanAirship/Platforms/Android/UAirshipPluginAndroid.cs]: White space + copyright. Ignore this file

File: Assets/UrbanAirship/Platforms/CustomEvent.cs [hunk 8]
```diff
@@ -1,165 +1,154 @@
-﻿/*
```
Comment (rlepinski) [inline Assets/UrbanAirship/Platforms/CustomEvent.cs]: White space + copyright. Ignore this file

File: Assets/UrbanAirship/Platforms/JsonArray.cs [hunk 9]
```diff
@@ -1,38 +1,32 @@
-﻿/*
```
Comment (rlepinski) [inline Assets/UrbanAirship/Platforms/JsonArray.cs]: White space + copyright. Ignore this file

File: Assets/UrbanAirship/Platforms/PushMessage.cs [hunk 10]
```diff
@@ -1,86 +1,80 @@
-﻿/*
- Copyright 2017 Urban Airship and Contributors
```
Comment (rlepinski) [inline Assets/UrbanAirship/Platforms/PushMessage.cs]: White space + copyright. Ignore this file

File: Assets/UrbanAirship/Platforms/TagGroupEditor.cs [hunk 11]
```diff
@@ -1,111 +1,99 @@
-﻿/*
```
Comment (rlepinski) [inline Assets/UrbanAirship/Platforms/TagGroupEditor.cs]: White space + copyright. Ignore this file

File: Assets/Scripts/UrbanAirshipBehaviour.cs [hunk 12]
```diff
@@ -1,67 +1,63 @@
-﻿/*
```
Comment (rlepinski) [inline Assets/Scripts/UrbanAirshipBehaviour.cs]: White space + copyright. Ignore this file

File: Assets/UrbanAirship/Editor/UAAssetPostprocessor.cs [hunk 13]
```diff
@@ -1,26 +1,23 @@
-﻿/*
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAAssetPostprocessor.cs]: White space + copyright. Ignore this file

File: Assets/UrbanAirship/Editor/UAMenu.cs [hunk 14]
```diff
@@ -1,46 +1,39 @@
-﻿/*
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAMenu.cs]: White space + copyright. Ignore this file

File: Assets/UrbanAirship/Editor/UAPostBuild.cs [hunk 15]
```diff
@@ -1,101 +1,94 @@
 /*
- Copyright 2017 Urban Airship and Contributors
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAPostBuild.cs]: White space + copyright. Ignore this file

File: README.md [hunk 16]
```diff
@@ -12,7 +12,7 @@ One of our engineers will verify receipt of the agreement before approving your
 
 ### Requirements
  - Unity 5
- - iOS - Xcode 8.3+ for iOS
+ - iOS - Xcode 9.0+ for iOS
```
Comment (BrianBatchelder) [inline README.md]: I'm not sure if our SDK will work in Xcode 9.0. You might want to play it safe and put 9.3.

File: unity-plugin/build.gradle [hunk 17]
```diff
@@ -56,39 +47,58 @@ task copyUnityClassesJar(type: Copy) {
...
+                root.appendNode('description', project.description)
+                root.appendNode('name', project.name)
+                root.appendNode('url', 'https://github.com/urbanairship/unity-plugin')
+                root.children().last() + pomConfig
+            }
+            repositories {
+                maven {
+                    url "$buildDir/m2repository"
+                }
+            }
```
Comment (BrianBatchelder) [inline unity-plugin/build.gradle]: Does this publish the library? If so, where does it get credentials?

Comment (rlepinski) [reply unity-plugin/build.gradle]: It publishes it to a local repo so i can package it with the plugin

**Comment by crow** [comment]
🍔

File: Assets/UrbanAirship/Editor/UAPreBuild.cs [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+#endif
+
+    class GenerateConfig {
+        public static void Apply (BuildTarget target) {
+            if (target == BuildTarget.iOS || target == BuildTarget.Android) {
+                UAConfig config = UAConfig.LoadConfig ();
+                if (!config.IsValid) {
+                    EditorUtility.DisplayDialog ("Urban Airship", "Urban Airship not configured. Set the app credentials in Window -> Urban Airship -> Settings", "OK");
```
Comment (BrianBatchelder) [inline Assets/UrbanAirship/Editor/UAPreBuild.cs]: Do you need to return here so you don't try to apply the invalid config?

**Comment by marc-scig** [comment]
👍 serialization is actually pretty tidy. I noticed that some (all?) of the reformatted files don't have newlines at the end, which github notoriously complains about, but I'm also not about to suggest bothering with that.

**Pull request merged**